orchestra/cmds/clean.py

```diff
@@ -14,21 +14,42 @@ def install_subcommand(sub_argparser: SubCommandParser):
         help="Remove build/source directories",
         parents=[execution_options],
     )
-    cmd_parser.add_argument("components", nargs="+", help="Name of the components to clean")
+    cmd_parser.add_argument("components", nargs="*", help="Name of the components to clean")
     cmd_parser.add_argument("--include-sources", "-s", action="store_true", help="Also delete source dir")
+    cmd_parser.add_argument("--all", action="store_true", help="Clean directories for all components")
+
+
+def clean_all(config: Configuration, args):
+    builds_dir = config.builds_dir
+    logger.info(f"Cleaning builds dir {builds_dir}")
+
+    if not args.pretend:
+        shutil.rmtree(builds_dir, ignore_errors=True)
+
+    if args.include_sources:
+        sources_dir = config.sources_dir
+        logger.info(f"Cleaning sources dir {sources_dir}")
+        if not args.pretend:
+            shutil.rmtree(sources_dir, ignore_errors=True)
 
 
 def handle_clean(args):
     config = Configuration(use_config_cache=args.config_cache)
 
-    builds = []
+    if bool(args.components) + bool(args.all) != 1:
+        logger.error("--all implies all components, do not specify any of them")
```

Comment:
I guess we need to fail globally here

Reply:
Yep, I totally forgot a return, thanks :)